apple/internal/partials/debug_symbols.bzl

```diff
@@ -300,13 +300,17 @@ def _debug_symbols_partial_impl(
 
     if platform_prerequisites.cpp_fragment:
         if platform_prerequisites.cpp_fragment.apple_generate_dsym:
+            dsym_output_filename = executable_name
+            if debug_discriminator:
+                dsym_output_filename += "_" + debug_discriminator
```

Comment:
I think if you make the following change to line 284

```diff
-debug_output_filename = bundle_name
+debug_output_filename = executable_name if executable_name else bundle_name 
```

https://github.com/bazelbuild/rules_apple/blob/a93ad60b28877b0cfa16abafc6bdd3fce0e66c76/apple/internal/partials/debug_symbols.bzl#L284-L286

there wouldn't be a need for a new `dsym_output_filename` parameter.

Reply:
I tried that change at first, but the output filename was
executable_name.app.dSYM/executable_name,
and the test (ios_application_custom_executable_name_dsyms_test) was expecting 
bundle_name.app.dSYM/executable_name
so I tried to separate the bundle name and filename.

Is executable_name.app.dSYM/executable_name OK too?

Also, I checked that the executable_name here is never None because we always pass the result of bundling_support.executable_name to this function, and basically it's equal to `executable_name if executable_name else bundle_name`

Reply:
Sorry I didn't realize that.

> Is executable_name.app.dSYM/executable_name OK as output?

I think `bundle_name.app.dSYM/executable_name` is the expected output.